### PR TITLE
Export servers

### PR DESCRIPTION
**Pull request by Georgehe4**
Export changes & service assets now in line with dart_services.

@devoncarew @lukechurch

### CODE REVIEW

**Comment by Georgehe4** [comment]
#557

File: lib/embed/embed.dart [hunk 1]
```diff
@@ -271,9 +273,11 @@ class PlaygroundMobile {
   }
 
   void _export() {
-    window.open(
-        "/index.html?dart=${Uri.encodeQueryComponent(context.dartSource)}&html=${Uri.encodeQueryComponent(context.htmlSource)}&css=${Uri.encodeQueryComponent(context.cssSource)}",
-        "DartPad");
+    P_dbservicesApi f = deps[P_dbservicesApi];
```
Comment (devoncarew) [inline lib/embed/embed.dart]: Instead of `f`, something like `dbServices`?

Comment (devoncarew) [inline lib/embed/embed.dart]: The auto-gened name - `P_dbservicesApi` - is very strange to look at.

File: lib/embed/embed.dart [hunk 2]
```diff
@@ -271,9 +273,11 @@ class PlaygroundMobile {
...
+    //TODO: Get key URL from server
```
Comment (devoncarew) [inline lib/embed/embed.dart]: Can remove this TODO:?

File: lib/playground.dart [hunk 3]
```diff
@@ -34,7 +34,7 @@ import 'sharing/mutable_gist.dart';
 import 'src/ga.dart';
 import 'src/summarize.dart';
 import 'src/util.dart';
-
+import 'services/_dbservices.dart';
```
Comment (devoncarew) [inline lib/playground.dart]: nit: need a line after the imports section

File: lib/playground.dart [hunk 4]
```diff
@@ -196,7 +196,18 @@ class Playground implements GistContainer, GistController {
       blankGist.getFile('index.html').content = url.queryParameters['html'];
       blankGist.getFile('styles.css').content = url.queryParameters['css'];
       editableGist.setBackingGist(blankGist);
-    } else if (_gistStorage.hasStoredGist && _gistStorage.storedId == null) {
+    } 
+    else if (url.hasQuery &&
```
Comment (devoncarew) [inline lib/playground.dart]: nit: here and below, the `else` should be on the same line as the `}`

File: lib/playground.dart [hunk 5]
```diff
@@ -196,7 +196,18 @@ class Playground implements GistContainer, GistController {
...
+        url.queryParameters['export'] != null) {
+      P_dbservicesApi f = deps[P_dbservicesApi];
+      Gist blankGist = createSampleGist();
+      f.returnContent(key:url.queryParameters['export']).then((onValue) {
```
Comment (devoncarew) [inline lib/playground.dart]: `key:url...` => `key: url...`

File: lib/playground.dart [hunk 6]
```diff
@@ -196,7 +196,18 @@ class Playground implements GistContainer, GistController {
       blankGist.getFile('index.html').content = url.queryParameters['html'];
       blankGist.getFile('styles.css').content = url.queryParameters['css'];
       editableGist.setBackingGist(blankGist);
-    } else if (_gistStorage.hasStoredGist && _gistStorage.storedId == null) {
+    } 
+    else if (url.hasQuery &&
+        url.queryParameters['export'] != null) {
```
Comment (devoncarew) [inline lib/playground.dart]: We had talked about using the `id` param like regular github gists, just running some heuristics on the id to see if it was a special export id. Not ciritical, but if we don't do it in this PR, we should add a TODO: to explore it later.

**Comment by devoncarew** [comment]
Just to note, we want to make sure that the services PR lands before we land this, and the new service is available on the backend.

**Comment by Georgehe4** [comment]
We're also changing some of the function names on the server end, so this PR will be updated after the server changes are finalized.

**Comment by lukechurch** [comment]
@Georgehe4 Good to close this?

**Comment by Georgehe4** [comment]
Closing.

**Pull request closed without merging**